src/Common/src/Interop/Ole32/Interop.GPStream.cs

```
@@ -0,0 +1,238 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Ole32
+    {
+        public sealed class GPStream : IStream
+        {
+            private Stream _dataStream;
+
+            // to support seeking ahead of the stream length...
+            private long _virtualPosition = -1;
+
+            internal GPStream(Stream stream, bool makeSeekable = true)
+            {
+                if (makeSeekable && !stream.CanSeek)
+                {
+                    // Copy to a memory stream so we can seek
+                    MemoryStream memoryStream = new MemoryStream();
+                    stream.CopyTo(memoryStream);
+                    _dataStream = memoryStream;
```

This is dangerous, if you have a non-seekable writable stream you redirect writing to a buffer and then drop any changes written by COM silently instead of forwarding them to the source stream. I don't know how relevant it is for WinForms, but WPF is having a lot of trouble with this because writing compressed streams can trigger this scenario (Desktop Framework had a buffered ZIP API while .NET Core produces non-seekable streams for both reading and writing).

Possible solutions:
- set a flag to remember you are buffering and reject writes from COM (since they are silently dropped anyways)
- actually implement writeback to the original stream

If the old implementation had the same behavior (dropping writes if the source stream is non-seekable) then you can of course ignore that comment since existing code can't depend on the behavior.

---

Fixed to be the same as old behaviour

---

Yeah I saw that, looks good, but github marked the comment as outdated rather than resolved and I don't see any option to change that. I can hide it but I don't think thats the same as marking resolved seen in on ther PRs. Am I missing anything or should I just hide my resolved comments which aren't auto-marked properly?